### ISSUE TITLE
Autocomplete of complex typedefs doesn't use the typedef

### ISSUE THREAD

**Issue by szotp** [dart-lang/sdk#43667]
Consider this code:
```dart
typedef Closure = T Function<T>(T input);

abstract class Base {
  Closure get closure;
}

class Concrete extends Base {}
```
Concrete is missing the getter, soI will now let IDE autocomplete it, and the result would be this:
```dart
class Concrete extends Base {
  @override
  // TODO: implement closure
  T Function<T>(T input) get closure => throw UnimplementedError();
}
```
For some reason, Closure got expanded into its raw form. I would expect the result to be this, as with non generic closures:
```dart
class Concrete extends Base {
  @override
  // TODO: implement closure
  Closure get closure => throw UnimplementedError();
}
```
Dart SDK version: 2.10.0 (stable) (Mon Sep 28 09:21:23 2020 +0200) on "macos_x64"

EDIT: seems like it affects other kinds of IDE helpers, like hover documentation.

**Comment by bwilkerson** [comment]
The analyzer doesn't currently provide us with enough information to be able to use the name of the typedef, though I believe that it should. I'm adding a failing test for this case (https://dart-review.googlesource.com/c/sdk/+/165981).

**Comment by scheglov** [comment]
Yes, the expected solution here is that the `FunctionType` produced from `Closure` keeps has its `element` pointing the `Closure`, and has `typeArguments` (empty in this case). But apparently we lose `element` and `typeArguments` somewhere on the way.

**Comment by scheglov** [comment]
I implemented it during 62f73a5bd9d5a03e2e690736b4832230adb60153 or close to it.
<img width="991" alt="image" src="https://github.com/dart-lang/sdk/assets/384794/11aa6615-dc60-42c5-9e62-3ad81b0395e7">
